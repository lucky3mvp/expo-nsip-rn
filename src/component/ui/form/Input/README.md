---
title: Input
subTitle: 输入框
---

# 输入框

### 组件描述
- 文本输入框


### 示例代码

```html
<Input label='错误状态' autoFocus error defaultValue='使用非受控属性，带错误状态，且自动聚焦'/>
<Input label='使用非受控属性' clear placeholder='带删除按钮的输入框'/>
<Input label='使用受控属性' clear value={this.state.inputValue} onChange={v => {
	this.setState({
		inputValue: v
	})
}}/>
<Input last defaultValue='最后一个，无底边框'/>
```

## API

### props列表

属性 | 说明 | 类型 | 默认值
----|-----|------|------
| last | 是否是一组输入框里的最后一个输入框, 默认输入框有 borderBottom, 但最后一个没有 | boolean | false |
| label | 输入框label | string or element | - |
| type | 输入框类型 | string | text |
| keyboardType | 键盘类型，不传则参考type自动判断键盘类型 | string | - |
| placeholder | 无输入文字时显示的提示文案 | string | - |
| clear | 是否显示清楚按钮 | boolean | false |
| value | 受控属性：需配合 onChange 使用更新数据 value | string | - |
| defaultValue | 非控属性 | string | '' |
| editable | 输入框是否可编辑 | boolean | true |
| maxLength | 最大字符数 | number | - |
| autoFormat | 是否自动格式化字符串 | boolean | true |
| format | 格式化字符串函数，在 autoFormat 为 true 时生效，不传则按照组件默认的方式格式化字符串，目前默认提供了‘银行卡’和‘手机号’两种字符串的格式化| function | - |
| autoFocus | 初始化时输入框是否获得焦点 | boolean | false |
| focused | 输入框存在期是否获得焦点的受控属性 | boolean | false |
| onChange | 文本变化回调，参数为输入框值和Input组件对象 | function(value, comp){} | - |
| onBlur | 输入框失去焦点回调，参数为输入框值 | function(value){} | - |
| onFocus | 输入框获得焦点回调，参数为输入框值 | function(value){} | - |
| extra | 输入框后缀信息 | string or element | - |
| onExtraClick | 后缀点击回调 | function(){} | - |
| error | 是否显示错误提示信息 | boolean | false |
| errorColor | 错误提示信息颜色 | string | '#f50'|
| errorIconFamily | 错误提示信息图标库 | string | - |
| errorIconName | 错误提示信息图标 | string | exclamation-circle |
| onErrorClick | 错误提示信息点击回调 | function(){} | - |

### 样式对象styles

属性 | 说明 | 适用类型
----|-----|------
| container | 容器样式 | View |
| label | label 类型为字符串时样式 | Text |
| input | 输入框样式 | TextInput |
| extra | 后缀信息为 string 时的样式 | Text |
| error | 错误信息容器的样式 | View |
| errorIcon | 错误图标的样式 | Icon |