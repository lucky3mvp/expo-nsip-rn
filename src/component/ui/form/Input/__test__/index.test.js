import React from 'react'
import Renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Input from '../index'

describe('Input Tests', () => {
	it('renders correctly', () => {
		const tree = Renderer.create((
			<Input label='label' placeholder='placeholder'/>
		)).toJSON();
		
		expect(tree).toMatchSnapshot();
	});

	it('children length test', () => {
		const wrapperWithLabel = shallow(
			<Input label='label' />
		);
		const wrapperWithExtra = shallow(
			<Input extra='extra' />
		);
		const wrapperWithAll = shallow(
			<Input label='label' extra='extra' error />
		);
		
		expect(wrapperWithLabel.children()).toHaveLength(2); // label & input
		expect(wrapperWithExtra.children()).toHaveLength(2); // extra & input
		expect(wrapperWithAll.children()).toHaveLength(4); // label & extra & input & error
	});
})
