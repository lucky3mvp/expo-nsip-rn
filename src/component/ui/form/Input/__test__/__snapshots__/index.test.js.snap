// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Tests renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderBottomColor": "#999",
        "borderBottomWidth": 0.5,
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 0,
        "marginLeft": 10,
        "marginTop": 0,
        "paddingRight": 10,
      },
      Object {
        "borderBottomWidth": 0.5,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333",
        "fontSize": 14,
        "marginRight": 5,
        "textAlignVertical": "center",
        "width": 60,
      }
    }
  >
    label
  </Text>
  <TextInput
    clearButtonMode="never"
    defaultValue=""
    editable={true}
    keyboardType="default"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="placeholder"
    secureTextEntry={false}
    style={
      Array [
        Object {
          "backgroundColor": "#FFF",
          "flex": 1,
          "fontSize": 14,
          "height": 48,
        },
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
